src/common/transformations/include/transformations/common_optimizations/division_by_zero_fp16_resolver.hpp

```diff
@@ -0,0 +1,35 @@
+// Copyright (C) 2018-2021 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#pragma once
+
+#include <utility>
+#include <memory>
+
+#include <transformations_visibility.hpp>
+#include <ngraph/pass/graph_rewrite.hpp>
+#include "ngraph/pattern/matcher.hpp"
+
+namespace ov {
+namespace pass {
+
+class TRANSFORMATIONS_API DivisionByZeroFP16Resolver;
+
+}  // namespace pass
+}  // namespace ov
+
+/**
+ * @ingroup ie_transformation_common_api
+ * @brief: clamps eps into fp16 minimal normalized value in input_1/Maximum(input_2, eps); input_1/Add(input_2, eps);
```

Comment:
nit: looks like this is not al patterns that are supported. Better to keep patterns description in cpp.